lib/src/cartesian/utils/transformInputData.ts

```diff
@@ -139,11 +139,14 @@ export const transformInputData = <
   });
 
   yKeys.forEach((yKey) => {
-    y[yKey].i = data.map((datum) => asNumber(datum[yKey]));
-    y[yKey].o = data.map((datum) => yScale(asNumber(datum[yKey])));
+    // Filter out non-numerical values (i.e. if the bar shouldn't exist)
+    const filtered = data.filter((datum) => typeof datum[yKey] === "number");
+
+    y[yKey].i = filtered.map((datum) => asNumber(datum[yKey]));
+    y[yKey].o = filtered.map((datum) => yScale(asNumber(datum[yKey])));
```

Comment:
I think this will cause some issues, specifically with touch gestures. The way the data is structured, it's expected that `ix`, `ox`, `y[key].i` and `y[key].o` all have the same length. For touch gestures, an array index is used to look up corresponding x/y values/coordinates, so if those arrays end up with different lengths, touch gesture state will break.

A way around that would be to filter the raw data if any of the `y` values are "invalid", but I'm not sure if that's bulletproof either. Because if you have something like `yKeys={["low", "high"]}` and the `low` data list is totally valid, but `low` data list is missing a some values – if you throw away data points based on values from `high`, you might be accidentally throwing away `low` data points that you're interested in.

All of that to say, I don't think this will work quite correctly across all use-cases – but I'm wondering if instead we can handle these cases in the bar/line/area/etc level.

Reply:
Thanks for taking the time to review! And I appreciate the work you did to fix it in the pull request.

I think you came up with a good solution, mainly removing the duplicated calls with asNumber since it produced the weird behavior.

Reply:
I love the work you guys are doing here. I think this is one of the more performant graphing libraries I've used. I've tried most of the popular ones now and even had my app working with the original victory native, but was doing a bit of a re-write to add some features and decided it was worth it to give this a go.

Thanks for all your hard work!